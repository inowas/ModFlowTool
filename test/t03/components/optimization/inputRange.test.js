import InputRange from '../../../../src/t03/components/optimization/inputRange';
import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

describe('Component renders as expected', () => {
    it('Renders correctly', () => {
        const rendered = renderer.create(
            <InputRange
                name="test"
                from={0}
                to={100}
                label="Test"
                label_from="min"
                label_to="max"
                onChange={() => {return true}}
            />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});

test('Return onChange after changing values and onBlur', () => {
    const doneChange = jest.fn();
    const wrapper = mount(
        <InputRange
            name="test"
            from={0}
            to={100}
            label="Test"
            label_from="min"
            label_to="max"
            onChange={doneChange}
        />
    );
    const input1 = wrapper.find('input[name="from"]').hostNodes();
    input1.simulate('change', {target: {name: 'from', value: 20}});
    const input2 = wrapper.find('input[name="to"]').hostNodes();
    input2.simulate('change', {target: {name: 'to', value: 80}});
    input2.simulate('blur');
    expect(doneChange).toBeCalledWith({"from": 20, "name": "test", "to": 80});
});