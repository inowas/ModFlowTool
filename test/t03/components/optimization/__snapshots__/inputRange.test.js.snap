// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected Renders correctly 1`] = `
<div
  className="equal width fields"
>
  <div
    className="field"
  >
    <label
      data-radium={true}
    >
      Test
       
      min
    </label>
    <div
      className="field"
    >
      <div
        className="ui input"
        style={
          Object {
            "padding": "0",
          }
        }
      >
        <input
          name="from"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="from ="
          type="number"
          value="0"
        />
      </div>
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      max
    </label>
    <div
      className="field"
    >
      <div
        className="ui input"
        style={
          Object {
            "padding": "0",
          }
        }
      >
        <input
          name="to"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="to ="
          type="number"
          value="100"
        />
      </div>
    </div>
  </div>
</div>
`;
